### PR TITLE
Make times BJD-TDB immediately, remove later conversion

### PR DESCRIPTION
**Pull request by mikeprimm**
I'm interested in feedback here, but let me outline the theory and motive:  the existing logic for producing the 'times' versus 'jd_times' arrays for the frame times currently results in jd_times always being a Julian Date, while 'times' will be either a BJD-TDB OR a JD time (depending upon whether the FITS files happen to have or not have BJD headers).  Besides the concern of having those values be essentially of inconsistent definition, the result of this is that the times used in the model fits are (potentially) inconsistent with the time reference for the priors - specifically, the fit_lightcurve (https://github.com/rzellem/EXOTIC/blob/develop/exotic/exotic.py#L1442) is called before the existing code ultimately 'normalizes' the times to BJD-TDB (https://github.com/rzellem/EXOTIC/blob/develop/exotic/exotic.py#L2249), which results in the processing of the times there (relative to computing phase and such from 'tmid', which IS always BJD-TDB) being inconsistent when that processing is being done.  I'm not sure how much effect this will have on the light curve fit (it might be more of an issue when tmid has very tight tolerances, for example), but it would seem to cause any inappropriate (and unnecessary) bias on the match relative to the priors.

In any case, the PR attempts to make 'times' be strongly defined as BJD-TDB as soon as the FITS are loaded, removing the need to 'fix' the values later (and ultimately resulting in the essentially the same processing costs, in either case), while assuring that all model fits are being done 'apples to apples' on times versus priors.

### CODE REVIEW

**Comment by jpl-jengelke** [comment]
In general, my preference is to keep a generic function that defaults to BJD-TDB unless it's not detected in the headers, in which case it would default to JD, or alternately do the conversion immediately. Because I don't know the original intent of this code, I defer to @pearsonkyle for more analysis.

File: .gitignore
Comment (jpl-jengelke) [inline .gitignore:134]: What is the `.project` used by?

Comment (mikeprimm) [reply .gitignore:134]: Eclipse work file - never part of source, and adding it to the .gitignore keeps it from being checked in

**Comment by pearsonkyle** [comment]
The logic makes sense to me. I think the BJD conversion was brought into the project after the light curve fitting part and so it happened to come after. I agree we should normalize the times prior to fitting the light curve otherwise we'd have to run another conversion on the mid-transit time to ensure it has the proper unit.

**Review by pearsonkyle — APPROVED** [review]

**Comment by rzellem** [comment]
@mikeprimm - can you please check and resolve this potential conflict in `exotic/exotic.py`? after that, I think we can *finally* merge! yay!!!

**Comment by mikeprimm** [comment]
Yep - some overlap, so I'll get it resolved now and do a quick retest.

**Review by rzellem — APPROVED** [review]

**Comment by mikeprimm** [comment]
OK - test run looks good - thanks for the merge!

**Pull request merged**
